cupy/_core/core.pyx

```diff
@@ -2215,24 +2217,34 @@ cpdef tuple assemble_cupy_compiler_options(tuple options):
 
     if not _is_hip:
         # CUDA Enhanced Compatibility
-        global _bundled_include
+        global _bundled_include, _headers_from_wheel_available
         if _bundled_include == '':
-            _cuda_major = runtime._getCUDAMajorVersion()
-            if _cuda_major == 11:
+            major, minor = nvrtc.getVersion()
+            if major == 11:
                 _bundled_include = 'cuda-11'
-            elif _cuda_major == 12:
-                _, minor = nvrtc.getVersion()
+            elif major == 12:
                 # TODO(leofang): update the upper bound when a new release
                 # is out
-                if minor < 2 or minor > 5:
+                if minor < 2:
                     _bundled_include = 'cuda-12'
-                else:
+                elif minor < 6:
                     _bundled_include = f'cuda-12.{minor}'
+                else:
+                    # Unsupported CUDA 12.x variant
+                    _bundled_include = None
             else:
                 # CUDA versions not yet supported.
                 _bundled_include = None
 
-        if _bundled_include is None and _cuda_path is None:
+            # Check if headers from cudart wheels are available.
+            wheel_dir_count = len(
+                _environment._get_include_dir_from_conda_or_wheel(major, minor))
```

Comment:
[The style check]( https://github.com/cupy/cupy/actions/runs/10465211580/job/28980078200?pr=8518#step:6:24 ) says this is 1 character over

```
cupy/_core/core.pyx:2241:80: E501 line too long (80 > 79 characters)
```

So let's wrap one

```suggestion
            wheel_dir_count = len(
                _environment._get_include_dir_from_conda_or_wheel(major, minor)
            )
```